src/main/java/com/conveyal/gtfs/GtfsGeometryCache.java

```diff
@@ -0,0 +1,91 @@
+package com.conveyal.gtfs;
+
+import com.conveyal.gtfs.model.Pattern;
+import com.conveyal.gtfs.model.Stop;
+import com.conveyal.r5.common.GeometryUtils;
+import org.locationtech.jts.geom.Coordinate;
+import org.locationtech.jts.geom.Envelope;
+import org.locationtech.jts.geom.LineString;
+import org.locationtech.jts.geom.Point;
+import org.locationtech.jts.index.strtree.STRtree;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * This class maintains spatial indexes of data from GTFS feeds.
```

Comment:
It's important that this indexes only _geometries for display_ from the GTFS feeds and does not attempt to provide full indexing for everything available in the GTFS API. Will update Javadoc to clarify.